src/test/ui/lint/implicit_unsafe_autorefs.stderr

```diff
@@ -47,5 +47,53 @@ help: try using a raw pointer method instead; or if this reference is intentiona
 LL |     addr_of!((&(*ptr).field)[..l - 1])
    |              ++            +
 
-warning: 4 warnings emitted
+warning: implicit auto-ref creates a reference to a dereference of a raw pointer
+  --> $DIR/implicit_unsafe_autorefs.rs:31:9
+   |
+LL |     _ = (*a)[0].len();
+   |         ^^^^^^^
+   |
+   = note: creating a reference requires the pointer to be valid and imposes aliasing requirements
+help: try using a raw pointer method instead; or if this reference is intentional, make it explicit
+   |
+LL |     _ = (&(*a)[0]).len();
+   |         ++       +
+
+warning: implicit auto-ref creates a reference to a dereference of a raw pointer
+  --> $DIR/implicit_unsafe_autorefs.rs:35:9
+   |
+LL |     _ = (*a)[..1][0].len();
+   |         ^^^^
+   |
+   = note: creating a reference requires the pointer to be valid and imposes aliasing requirements
+help: try using a raw pointer method instead; or if this reference is intentional, make it explicit
+   |
+LL |     _ = (&(*a))[..1][0].len();
+   |         ++    +
+
+warning: implicit auto-ref creates a reference to a dereference of a raw pointer
+  --> $DIR/implicit_unsafe_autorefs.rs:40:18
+   |
+LL |     _ = addr_of!((*ptr).field);
+   |                  ^^^^^^
+   |
+   = note: creating a reference requires the pointer to be valid and imposes aliasing requirements
```

Comment:
Would it be possible to explain where the implicit reference comes from?
I imagine some staring at this and being like "it's just a field access, wtf". The error could say something like
- "a reference is implicitly created to match the method receiver type"
- "a reference is implicitly created because a deref coercion is being applied"
- "a reference is implicitly created because a user-defined indexing operation is being called"